test/celeritas/data/four-steel-slabs.root-dump.json

```diff
@@ -1,540 +1 @@
d>", "first" : -13, "second" : {"_typename" : "celeritas::ImportLoopingThreshold","threshold_trials" : 10,"important_energy" : 250}}, {"$pair" : "pair<int,celeritas::ImportLoopingThreshold>", "first" : -11, "second" : {"_typename" : "celeritas::ImportLoopingThreshold","threshold_trials" : 10,"important_energy" : 250}}, {"$pair" : "pair<int,celeritas::ImportLoopingThreshold>", "first" : 11, "second" : {"_typename" : "celeritas::ImportLoopingThreshold","threshold_trials" : 10,"important_energy" : 250}}, {"$pair" : "pair<int,celeritas::ImportLoopingThreshold>", "first" : 13, "second" : {"_typename" : "celeritas::ImportLoopingThreshold","threshold_trials" : 10,"important_energy" : 250}}, {"$pair" : "pair<int,celeritas::ImportLoopingThreshold>", "first" : 22, "second" : {"_typename" : "celeritas::ImportLoopingThreshold","threshold_trials" : 10,"important_energy" : 250}}],"max_substeps" : 1000},"optical_params" : {"_typename" : "celeritas::ImportOpticalParameters","scintillation_by_particle" : false,"wls_time_profile" : 0,"wls2_time_profile" : 0},"optical_models" : [],"optical_materials" : [],"units" : "cgs","optical_physics" : {"_typename" : "celeritas::inp::OpticalPhysics","cherenkov" : true,"scintillation" : true,"surfaces" : {"_typename" : "celeritas::inp::SurfacePhysics","materials" : [[]],"roughness" : {"_typename" : "celeritas::inp::RoughnessModels","polished" : [{"$pair" : "pair<celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>,celeritas::inp::NoRoughness>", "first" : {"_typename" : "celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>","value_" : 0}, "second" : {"_typename" : "celeritas::inp::NoRoughness"}}],"smear" : [],"gaussian" : []},"reflectivity" : {"_typename" : "celeritas::inp::ReflectivityModels","grid" : [],"fresnel" : [{"$pair" : "pair<celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>,celeritas::inp::FresnelReflection>", "first" : {"_typename" : "celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>","value_" : 0}, "second" : {"_typename" : "celeritas::inp::FresnelReflection"}}]},"interaction" : {"_typename" : "celeritas::inp::InteractionModels","dielectric_dielectric" : [{"$pair" : "pair<celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>,celeritas::inp::ReflectionForm>", "first" : {"_typename" : "celeritas::OpaqueId<celeritas::PhysSurface_,unsigned int>","value_" : 0}, "second" : {"_typename" : "celeritas::inp::ReflectionForm","specular_spike" : {"_typename" : "celeritas::inp::Grid","x" : [0, 1e308],"y" : [1, 1],"interpolation" : {"_typename" : "celeritas::inp::Interpolation","type" : 0,"order" : 1,"bc" : 2}},"specular_lobe" : {"_typename" : "celeritas::inp::Grid","x" : [0, 1e308],"y" : [0, 0],"interpolation" : {"_typename" : "celeritas::inp::Interpolation","type" : 0,"order" : 1,"bc" : 2}},"backscatter" : {"_typename" : "celeritas::inp::Grid","x" : [0, 1e308],"y" : [0, 0],"interpolation" : {"_typename" : "celeritas::inp::Interpolation","type" : 0,"order" : 1,"bc" : 2}}}}],"dielectric_metal" : []}}}}
```

Comment:
Is this what "compact" does? It's somewhat useful to see the changes as a diff rather than just getting a massive block of text...

Reply:
Oh. What about replacing the code in the ROOT comparer with code similar to orange/orangeinp/UnitProto.test.cc ? (Instead of telling you to update the file it could tell you to delete it and it would replace it on the next run.)